File: VB/Example/Controllers/MainController.vb
Imports System
Imports System.Linq
Imports System.Web.Mvc
Imports DevExpress.Xpo

<HandleError()> _
Public Class MainController
    Inherits Controller
    Public Function Index() As ActionResult
        Return View(GetData())
    End Function
    Public Function GridPartial() As ActionResult
        Return View("GridPartial", GetData())
    End Function
    Public Function GetData() As Object
        Dim session As Session = XpoHelper.GetNewSession()

        Dim qry As New XPQuery(Of Customer)(session)
        Dim list = _
         From c In qry _
         Select New With {Key c.Name, Key c.CompanyName}
        Return list.ToList()
    End Function
End Class

File: CS/Example/Controllers/MainController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using DevExpress.Xpo;

namespace Example.Controllers {
    public class MainController : Controller {
        public ActionResult Index() {
            return View(GetData());
        }
        public ActionResult GridPartial() {
            return View("GridPartial", GetData());
        }
        public object GetData() {
            Session session = XpoHelper.GetNewSession();

            XPQuery<Customer> qry = new XPQuery<Customer>(session);
            var list = from c in qry
                       select new { c.Name, c.CompanyName };
            return list.ToList();
        }
    }
}
